In [1]:
import pandas as pd
import numpy as np

# Загрузка датасета
df = pd.read_csv('all_sport_objects_with_infrastructure.csv')
df

sport_object_id                       sport_object_name  \
0       5348552838531230  Спортивный комплекс им. В.И. Алексеева   
1       5348552838531230  Спортивный комплекс им. В.И. Алексеева   
2       5348552838531230  Спортивный комплекс им. В.И. Алексеева   
3       5348552838531230  Спортивный комплекс им. В.И. Алексеева   
4       5348552838531230  Спортивный комплекс им. В.И. Алексеева   
...                  ...                                     ...   
15713  70000001109877497                      Vibora, падел-клуб   
15714  70000001109877497                      Vibora, падел-клуб   
15715  70000001109877497                      Vibora, падел-клуб   
15716  70000001109877497                      Vibora, падел-клуб   
15717  70000001109877497                      Vibora, падел-клуб   

           sport_object_address  sport_object_lat  sport_object_lon  \
0        Раевского проспект, 16         60.012929         30.361267   
1        Раевского проспект, 16         60.012929         30.361267   
2        Раевского проспект, 16         60.012929         30.361267   
3        Раевского проспект, 16         60.012929         30.361267   
4        Раевского проспект, 16         60.012929         30.361267   
...                         ...               ...               ...   
15713  Выборгское шоссе, 503 к1         60.098474         30.259764   
15714  Выборгское шоссе, 503 к1         60.098474         30.259764   
15715  Выборгское шоссе, 503 к1         60.098474         30.259764   
15716  Выборгское шоссе, 503 к1         60.098474         30.259764   
15717  Выборгское шоссе, 503 к1         60.098474         30.259764   

      sport_object_type  search_keyword  sport_object_total_infrastructure  \
0                другое  теннисный корт                                 50   
1                другое  теннисный корт                                 50   
2                другое  теннисный корт                                 50   
3                другое  теннисный корт                                 50   
4                другое  теннисный корт                                 50   
...                 ...             ...                                ...   
15713             паддл           падел                                 12   
15714             паддл           падел                                 12   
15715             паддл           падел                                 12   
15716             паддл           падел                                 12   
15717             паддл           падел                                 12   

                                                schedule  \
0      {"Fri": {"working_hours": [{"from": "06:00", "...   
1      {"Fri": {"working_hours": [{"from": "06:00", "...   
2      {"Fri": {"working_hours": [{"from": "06:00", "...   
3      {"Fri": {"working_hours": [{"from": "06:00", "...   
4      {"Fri": {"working_hours": [{"from": "06:00", "...   
...                                                  ...   
15713  {"Fri": {"working_hours": [{"from": "08:00", "...   
15714  {"Fri": {"working_hours": [{"from": "08:00", "...   
15715  {"Fri": {"working_hours": [{"from": "08:00", "...   
15716  {"Fri": {"working_hours": [{"from": "08:00", "...   
15717  {"Fri": {"working_hours": [{"from": "08:00", "...   

                                                 rubrics  ...  \
0      Фитнес-клубы | Массажист | Тренажёрные залы | ...  ...   
1      Фитнес-клубы | Массажист | Тренажёрные залы | ...  ...   
2      Фитнес-клубы | Массажист | Тренажёрные залы | ...  ...   
3      Фитнес-клубы | Массажист | Тренажёрные залы | ...  ...   
4      Фитнес-клубы | Массажист | Тренажёрные залы | ...  ...   
...                                                  ...  ...   
15713                                        Падел-корты  ...   
15714                                        Падел-корты  ...   
15715                                        Падел-корты  ...   
15716                          

In [2]:
unique = df['schedule'].unique()
unique

array(['{"Fri": {"working_hours": [{"from": "06:00", "to": "01:00"}]}, "Mon": {"working_hours": [{"from": "06:00", "to": "01:00"}]}, "Sat": {"working_hours": [{"from": "06:00", "to": "01:00"}]}, "Sun": {"working_hours": [{"from": "06:00", "to": "01:00"}]}, "Thu": {"working_hours": [{"from": "06:00", "to": "01:00"}]}, "Tue": {"working_hours": [{"from": "06:00", "to": "01:00"}]}, "Wed": {"working_hours": [{"from": "06:00", "to": "01:00"}]}}',
       '{"Fri": {"working_hours": [{"from": "06:30", "to": "23:00"}]}, "Mon": {"working_hours": [{"from": "06:30", "to": "23:00"}]}, "Sat": {"working_hours": [{"from": "07:30", "to": "23:00"}]}, "Sun": {"working_hours": [{"from": "07:30", "to": "23:00"}]}, "Thu": {"working_hours": [{"from": "06:30", "to": "23:00"}]}, "Tue": {"working_hours": [{"from": "06:30", "to": "23:00"}]}, "Wed": {"working_hours": [{"from": "06:30", "to": "23:00"}]}}',
       '{"Fri": {"working_hours": [{"from": "08:00", "to": "23:00"}]}, "Mon": {"working_hours": [{"from": "08:

In [3]:
# Создаем функцию для проверки условий
def check_schedule(schedule_value):

    if pd.isna(schedule_value):
        return 0

    schedule_str = str(schedule_value)

    has_00_00 = "00:00" in schedule_str
    has_to_24_00 = '"to": "24:00"' in schedule_str or "'to': '24:00'" in schedule_str

    return 1 if (has_00_00 and has_to_24_00) else 0

df['schedule'] = df['schedule'].apply(check_schedule)

In [4]:
df

sport_object_id                       sport_object_name  \
0       5348552838531230  Спортивный комплекс им. В.И. Алексеева   
1       5348552838531230  Спортивный комплекс им. В.И. Алексеева   
2       5348552838531230  Спортивный комплекс им. В.И. Алексеева   
3       5348552838531230  Спортивный комплекс им. В.И. Алексеева   
4       5348552838531230  Спортивный комплекс им. В.И. Алексеева   
...                  ...                                     ...   
15713  70000001109877497                      Vibora, падел-клуб   
15714  70000001109877497                      Vibora, падел-клуб   
15715  70000001109877497                      Vibora, падел-клуб   
15716  70000001109877497                      Vibora, падел-клуб   
15717  70000001109877497                      Vibora, падел-клуб   

           sport_object_address  sport_object_lat  sport_object_lon  \
0        Раевского проспект, 16         60.012929         30.361267   
1        Раевского проспект, 16         60.012929         30.361267   
2        Раевского проспект, 16         60.012929         30.361267   
3        Раевского проспект, 16         60.012929         30.361267   
4        Раевского проспект, 16         60.012929         30.361267   
...                         ...               ...               ...   
15713  Выборгское шоссе, 503 к1         60.098474         30.259764   
15714  Выборгское шоссе, 503 к1         60.098474         30.259764   
15715  Выборгское шоссе, 503 к1         60.098474         30.259764   
15716  Выборгское шоссе, 503 к1         60.098474         30.259764   
15717  Выборгское шоссе, 503 к1         60.098474         30.259764   

      sport_object_type  search_keyword  sport_object_total_infrastructure  \
0                другое  теннисный корт                                 50   
1                другое  теннисный корт                                 50   
2                другое  теннисный корт                                 50   
3                другое  теннисный корт                                 50   
4                другое  теннисный корт                                 50   
...                 ...             ...                                ...   
15713             паддл           падел                                 12   
15714             паддл           падел                                 12   
15715             паддл           падел                                 12   
15716             паддл           падел                                 12   
15717             паддл           падел                                 12   

       schedule                                            rubrics  ...  \
0             0  Фитнес-клубы | Массажист | Тренажёрные залы | ...  ...   
1             0  Фитнес-клубы | Массажист | Тренажёрные залы | ...  ...   
2             0  Фитнес-клубы | Массажист | Тренажёрные залы | ...  ...   
3             0  Фитнес-клубы | Массажист | Тренажёрные залы | ...  ...   
4             0  Фитнес-клубы | Массажист | Тренажёрные залы | ...  ...   
...         ...                                                ...  ...   
15713         0                                        Падел-корты  ...   
15714         0                                        Падел-корты  ...   
15715         0                                        Падел-корты  ...   
15716         0                                        Падел-корты  ...   
15717         0                                        Падел-корты  ...   

                                     infrastructure_name  \
0                                            Олимп, кафе   
1                                      Чайхона, ресторан   
2                                            Кебаб, кафе   
3                                       Южная-ночь, кафе   
4                         Пироговый дворик, кафе-пекарня   
...                                                  ...   
15713                              Kerama Marazzi, склад   
15714                

In [5]:
unique = df['sport_object_type'].unique()
unique

array(['другое', 'теннис', 'паддл', 'сквош'], dtype=object)

In [6]:
other_df = df[df['sport_object_type'] == 'другое']
result = other_df[['sport_object_type', 'search_keyword']]
result

sport_object_type  search_keyword
0                другое  теннисный корт
1                другое  теннисный корт
2                другое  теннисный корт
3                другое  теннисный корт
4                другое  теннисный корт
...                 ...             ...
15550            другое           падел
15551            другое           падел
15552            другое           падел
15553            другое           падел
15554            другое           падел

[6740 rows x 2 columns]

In [7]:
result1 = result['search_keyword'].unique()
result1

array(['теннисный корт', 'сквош корт', 'падел'], dtype=object)

In [8]:
def update_sport_type(row):
    keyword = row['search_keyword']
    if keyword == 'теннисный корт':
        return 'теннис'
    elif keyword == 'сквош корт':
        return 'сквош'
    elif keyword == 'падел':
        return 'паддл'
    else:
        # Оставляем текущее значение
        return row['sport_object_type']

df['sport_object_type'] = df.apply(update_sport_type, axis=1)
df

sport_object_id                       sport_object_name  \
0       5348552838531230  Спортивный комплекс им. В.И. Алексеева   
1       5348552838531230  Спортивный комплекс им. В.И. Алексеева   
2       5348552838531230  Спортивный комплекс им. В.И. Алексеева   
3       5348552838531230  Спортивный комплекс им. В.И. Алексеева   
4       5348552838531230  Спортивный комплекс им. В.И. Алексеева   
...                  ...                                     ...   
15713  70000001109877497                      Vibora, падел-клуб   
15714  70000001109877497                      Vibora, падел-клуб   
15715  70000001109877497                      Vibora, падел-клуб   
15716  70000001109877497                      Vibora, падел-клуб   
15717  70000001109877497                      Vibora, падел-клуб   

           sport_object_address  sport_object_lat  sport_object_lon  \
0        Раевского проспект, 16         60.012929         30.361267   
1        Раевского проспект, 16         60.012929         30.361267   
2        Раевского проспект, 16         60.012929         30.361267   
3        Раевского проспект, 16         60.012929         30.361267   
4        Раевского проспект, 16         60.012929         30.361267   
...                         ...               ...               ...   
15713  Выборгское шоссе, 503 к1         60.098474         30.259764   
15714  Выборгское шоссе, 503 к1         60.098474         30.259764   
15715  Выборгское шоссе, 503 к1         60.098474         30.259764   
15716  Выборгское шоссе, 503 к1         60.098474         30.259764   
15717  Выборгское шоссе, 503 к1         60.098474         30.259764   

      sport_object_type  search_keyword  sport_object_total_infrastructure  \
0                теннис  теннисный корт                                 50   
1                теннис  теннисный корт                                 50   
2                теннис  теннисный корт                                 50   
3                теннис  теннисный корт                                 50   
4                теннис  теннисный корт                                 50   
...                 ...             ...                                ...   
15713             паддл           падел                                 12   
15714             паддл           падел                                 12   
15715             паддл           падел                                 12   
15716             паддл           падел                                 12   
15717             паддл           падел                                 12   

       schedule                                            rubrics  ...  \
0             0  Фитнес-клубы | Массажист | Тренажёрные залы | ...  ...   
1             0  Фитнес-клубы | Массажист | Тренажёрные залы | ...  ...   
2             0  Фитнес-клубы | Массажист | Тренажёрные залы | ...  ...   
3             0  Фитнес-клубы | Массажист | Тренажёрные залы | ...  ...   
4             0  Фитнес-клубы | Массажист | Тренажёрные залы | ...  ...   
...         ...                                                ...  ...   
15713         0                                        Падел-корты  ...   
15714         0                                        Падел-корты  ...   
15715         0                                        Падел-корты  ...   
15716         0                                        Падел-корты  ...   
15717         0                                        Падел-корты  ...   

                                     infrastructure_name  \
0                                            Олимп, кафе   
1                                      Чайхона, ресторан   
2                                            Кебаб, кафе   
3                                       Южная-ночь, кафе   
4                         Пироговый дворик, кафе-пекарня   
...                                                  ...   
15713                              Kerama Marazzi, склад   
15714                

In [9]:
unique = df['sport_object_type'].unique()
unique

array(['теннис', 'сквош', 'паддл'], dtype=object)

In [10]:
df.drop(columns=['search_keyword', 'rubrics'], inplace=True)
df

sport_object_id                       sport_object_name  \
0       5348552838531230  Спортивный комплекс им. В.И. Алексеева   
1       5348552838531230  Спортивный комплекс им. В.И. Алексеева   
2       5348552838531230  Спортивный комплекс им. В.И. Алексеева   
3       5348552838531230  Спортивный комплекс им. В.И. Алексеева   
4       5348552838531230  Спортивный комплекс им. В.И. Алексеева   
...                  ...                                     ...   
15713  70000001109877497                      Vibora, падел-клуб   
15714  70000001109877497                      Vibora, падел-клуб   
15715  70000001109877497                      Vibora, падел-клуб   
15716  70000001109877497                      Vibora, падел-клуб   
15717  70000001109877497                      Vibora, падел-клуб   

           sport_object_address  sport_object_lat  sport_object_lon  \
0        Раевского проспект, 16         60.012929         30.361267   
1        Раевского проспект, 16         60.012929         30.361267   
2        Раевского проспект, 16         60.012929         30.361267   
3        Раевского проспект, 16         60.012929         30.361267   
4        Раевского проспект, 16         60.012929         30.361267   
...                         ...               ...               ...   
15713  Выборгское шоссе, 503 к1         60.098474         30.259764   
15714  Выборгское шоссе, 503 к1         60.098474         30.259764   
15715  Выборгское шоссе, 503 к1         60.098474         30.259764   
15716  Выборгское шоссе, 503 к1         60.098474         30.259764   
15717  Выборгское шоссе, 503 к1         60.098474         30.259764   

      sport_object_type  sport_object_total_infrastructure  schedule  \
0                теннис                                 50         0   
1                теннис                                 50         0   
2                теннис                                 50         0   
3                теннис                                 50         0   
4                теннис                                 50         0   
...                 ...                                ...       ...   
15713             паддл                                 12         0   
15714             паддл                                 12         0   
15715             паддл                                 12         0   
15716             паддл                                 12         0   
15717             паддл                                 12         0   

      infrastructure_type                                infrastructure_name  \
0                    кафе                                        Олимп, кафе   
1                    кафе                                  Чайхона, ресторан   
2                    кафе                                        Кебаб, кафе   
3                    кафе                                   Южная-ночь, кафе   
4                    кафе                     Пироговый дворик, кафе-пекарня   
...                   ...                                                ...   
15713                кафе                              Kerama Marazzi, склад   
15714      торговый_центр                     Китай город, торговый комплекс   
15715         супермаркет  Максидом, гипермаркет строительно-отделочных м...   
15716           остановка                         Pereobuvka, автомастерская   
15717                офис                     Китай город, торговый комплекс   

          infrastructure_address  infrastructure_lat  infrastructure_lon  \
0         Раевского проспект, 16           60.013074           30.361556   
1      Тихорецкий проспект, 2 к2           60.010418           30.368360   
2      Тихорецкий проспект, 1 к1           60.010388           30.369832   
3             улица Курчатова, 9           60.007857           30.360255   
4      Политехническая улица, 31           60.009633           30.369327   
...                          ...                 ...     

In [11]:
df['sport_object_type'] = df['sport_object_type'].replace('паддл', 'падел')
df

sport_object_id                       sport_object_name  \
0       5348552838531230  Спортивный комплекс им. В.И. Алексеева   
1       5348552838531230  Спортивный комплекс им. В.И. Алексеева   
2       5348552838531230  Спортивный комплекс им. В.И. Алексеева   
3       5348552838531230  Спортивный комплекс им. В.И. Алексеева   
4       5348552838531230  Спортивный комплекс им. В.И. Алексеева   
...                  ...                                     ...   
15713  70000001109877497                      Vibora, падел-клуб   
15714  70000001109877497                      Vibora, падел-клуб   
15715  70000001109877497                      Vibora, падел-клуб   
15716  70000001109877497                      Vibora, падел-клуб   
15717  70000001109877497                      Vibora, падел-клуб   

           sport_object_address  sport_object_lat  sport_object_lon  \
0        Раевского проспект, 16         60.012929         30.361267   
1        Раевского проспект, 16         60.012929         30.361267   
2        Раевского проспект, 16         60.012929         30.361267   
3        Раевского проспект, 16         60.012929         30.361267   
4        Раевского проспект, 16         60.012929         30.361267   
...                         ...               ...               ...   
15713  Выборгское шоссе, 503 к1         60.098474         30.259764   
15714  Выборгское шоссе, 503 к1         60.098474         30.259764   
15715  Выборгское шоссе, 503 к1         60.098474         30.259764   
15716  Выборгское шоссе, 503 к1         60.098474         30.259764   
15717  Выборгское шоссе, 503 к1         60.098474         30.259764   

      sport_object_type  sport_object_total_infrastructure  schedule  \
0                теннис                                 50         0   
1                теннис                                 50         0   
2                теннис                                 50         0   
3                теннис                                 50         0   
4                теннис                                 50         0   
...                 ...                                ...       ...   
15713             падел                                 12         0   
15714             падел                                 12         0   
15715             падел                                 12         0   
15716             падел                                 12         0   
15717             падел                                 12         0   

      infrastructure_type                                infrastructure_name  \
0                    кафе                                        Олимп, кафе   
1                    кафе                                  Чайхона, ресторан   
2                    кафе                                        Кебаб, кафе   
3                    кафе                                   Южная-ночь, кафе   
4                    кафе                     Пироговый дворик, кафе-пекарня   
...                   ...                                                ...   
15713                кафе                              Kerama Marazzi, склад   
15714      торговый_центр                     Китай город, торговый комплекс   
15715         супермаркет  Максидом, гипермаркет строительно-отделочных м...   
15716           остановка                         Pereobuvka, автомастерская   
15717                офис                     Китай город, торговый комплекс   

          infrastructure_address  infrastructure_lat  infrastructure_lon  \
0         Раевского проспект, 16           60.013074           30.361556   
1      Тихорецкий проспект, 2 к2           60.010418           30.368360   
2      Тихорецкий проспект, 1 к1           60.010388           30.369832   
3             улица Курчатова, 9           60.007857           30.360255   
4      Политехническая улица, 31           60.009633           30.369327   
...                          ...                 ...     

In [12]:
# Импортируем датафрейм по аналитике районов

# Загрузка датасета
df2 = pd.read_csv('full_cluster_analysis.csv')
df2

Кластер  Тип_кластера_района  Порядковый_номер_в_кластере  \
0         0      МАССОВЫЕ РАЙОНЫ                            1   
1         0      МАССОВЫЕ РАЙОНЫ                            2   
2         0      МАССОВЫЕ РАЙОНЫ                            3   
3         0      МАССОВЫЕ РАЙОНЫ                            4   
4         0      МАССОВЫЕ РАЙОНЫ                            5   
5         0      МАССОВЫЕ РАЙОНЫ                            6   
6         1        СМЕШАННЫЙ ТИП                            1   
7         1        СМЕШАННЫЙ ТИП                            2   
8         1        СМЕШАННЫЙ ТИП                            3   
9         1        СМЕШАННЫЙ ТИП                            4   
10        1        СМЕШАННЫЙ ТИП                            5   
11        2  ПРИОРИТЕТНЫЕ РАЙОНЫ                            1   
12        2  ПРИОРИТЕТНЫЕ РАЙОНЫ                            2   
13        2  ПРИОРИТЕТНЫЕ РАЙОНЫ                            3   
14        2  ПРИОРИТЕТНЫЕ РАЙОНЫ                            4   
15        3       НИШЕВЫЕ РАЙОНЫ                            1   
16        3       НИШЕВЫЕ РАЙОНЫ                            2   
17        4       НИШЕВЫЕ РАЙОНЫ                            1   

           Населенный_пункт  Плотность_населения  Отклонение_плотности_%  \
0          Выборгский район              4620.91              -28.234264   
1           Кировский район              7162.79               11.242784   
2   Красногвардейский район              6554.52                1.795956   
3      Красносельский район              4711.81              -26.822528   
4             Невский район              9008.82               39.912830   
5         Фрунзенский район             11102.61               72.430749   
6          Колпинский район              1798.99              -72.060516   
7       Кронштадтский район              2273.37              -64.693086   
8           Курортный район               308.41              -95.210192   
9      Петродворцовый район              1251.57              -80.562304   
10         Пушкинский район              1065.04              -83.459237   
11     Адмиралтейский район             11451.01               77.841628   
12   Василеостровский район             12558.86               95.047258   
13        Калининский район             13611.45              111.394665   
14        Центральный район             11472.71               78.178643   
15         Московский район              4655.63              -27.695040   
16         Приморский район              6303.48               -2.102859   
17      Петроградский район              5987.87               -7.004487   

    Зарплата  Отклонение_зарплаты_%  Население  Соотношение_М_Ж  \
0     131990              -0.581866     533808            0.812   
1     129717              -2.293946     339946            0.837   
2     126337              -4.839846     369347            0.795   
3     108531             -18.251765     426372            0.804   
4     120834              -8.984841     546475            0.818   
5     119480             -10.004708     416570            0.796   
6     125218              -5.682704     183947            0.817   
7     108831             -18.025798      44399            0.895   
8     118117             -11.031353      82713            0.887   
9     117752             -11.306280     134018            0.876   
10    110107             -17.064683     255609            0.878   
11    134590               1.376518     158253            0.828   
12    148492              11.847849     209733            0.816   
13    145695               9.741079     546908            0.813   
14    159976              20.497882     203870            0.809   
15    163332              23.025704     340187            0.776   
16    166640              25.517371     692753            0.834   
17    154086              16.061388     117003            0.946   

    Средняя_плотность Класс_плотности  Средняя_за

In [13]:
df['district'].unique()

array(['Выборгский район', 'Невский район', 'Фрунзенский район',
       'Пушкинский район', '0', 'Московский район', 'Приморский район',
       'Кировский район', 'Красногвардейский район',
       'Адмиралтейский район', 'Василеостровский район',
       'Петроградский район', 'Калининский район', 'Красносельский район',
       'Курортный район'], dtype=object)

In [14]:
# Мерджим 2 датафрема

df_final = pd.merge(df, df2,
                  left_on='district',
                  right_on='Населенный_пункт',
                  how='outer')
df_final

sport_object_id                 sport_object_name  \
0         7.000000e+16  Fitness Life Европа, фитнес-клуб   
1         7.000000e+16  Fitness Life Европа, фитнес-клуб   
2         7.000000e+16  Fitness Life Европа, фитнес-клуб   
3         7.000000e+16  Fitness Life Европа, фитнес-клуб   
4         7.000000e+16  Fitness Life Европа, фитнес-клуб   
...                ...                               ...   
15717     7.000000e+16            Padel star, падел клуб   
15718     7.000000e+16            Padel star, падел клуб   
15719     7.000000e+16            Padel star, падел клуб   
15720     7.000000e+16            Padel star, падел клуб   
15721              NaN                               NaN   

              sport_object_address  sport_object_lat  sport_object_lon  \
0      Европейский проспект, 2 ст1         59.898325         30.513494   
1      Европейский проспект, 2 ст1         59.898325         30.513494   
2      Европейский проспект, 2 ст1         59.898325         30.513494   
3      Европейский проспект, 2 ст1         59.898325         30.513494   
4      Европейский проспект, 2 ст1         59.898325         30.513494   
...                            ...               ...               ...   
15717              Фучика, 2 лит А         59.878109         30.358340   
15718              Фучика, 2 лит А         59.878109         30.358340   
15719              Фучика, 2 лит А         59.878109         30.358340   
15720              Фучика, 2 лит А         59.878109         30.358340   
15721                          NaN               NaN               NaN   

      sport_object_type  sport_object_total_infrastructure  schedule  \
0                теннис                               89.0       0.0   
1                теннис                               89.0       0.0   
2                теннис                               89.0       0.0   
3                теннис                               89.0       0.0   
4                теннис                               89.0       0.0   
...                 ...                                ...       ...   
15717             падел                               82.0       0.0   
15718             падел                               82.0       0.0   
15719             падел                               82.0       0.0   
15720             падел                               82.0       0.0   
15721               NaN                                NaN       NaN   

      infrastructure_type                          infrastructure_name  ...  \
0                    кафе                                    Toki-Poki  ...   
1                    кафе        Два Топора, ресторан быстрого питания  ...   
2                    кафе               Шаверно, кафе быстрого питания  ...   
3                    кафе             Хачо и Пури, грузинский ресторан  ...   
4                    кафе                         Додо Пицца, пиццерия  ...   
...                   ...                                          ...  ...   
15717                офис  Континент, торгово-развлекательный комплекс  ...   
15718                офис                                     СберБанк  ...   
15719                офис                                   Почта Банк  ...   
15720                офис                Бухарестская 24, бизнес-центр  ...   
15721                 NaN                                          NaN  ...   

      Отклонение_плотности_%  Зарплата  Отклонение_зарплаты_%  Население  \
0                        NaN       NaN                    NaN        NaN   
1                        NaN       NaN                    NaN        NaN   
2                        NaN       NaN                    NaN        NaN   
3                        NaN       NaN                    NaN        NaN   
4                        NaN       NaN                    NaN        NaN   
...                      ...       ...                    ...        ...   
15717              72.430749  119480.0             

In [15]:
df_final.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 15722 entries, 0 to 15721
Data columns (total 34 columns):
 #   Column                             Non-Null Count  Dtype  
---  ------                             --------------  -----  
 0   sport_object_id                    15718 non-null  float64
 1   sport_object_name                  15718 non-null  object 
 2   sport_object_address               15640 non-null  object 
 3   sport_object_lat                   15718 non-null  float64
 4   sport_object_lon                   15718 non-null  float64
 5   sport_object_type                  15718 non-null  object 
 6   sport_object_total_infrastructure  15718 non-null  float64
 7   schedule                           15718 non-null  float64
 8   infrastructure_type                15718 non-null  object 
 9   infrastructure_name                15718 non-null  object 
 10  infrastructure_address             14856 non-null  object 
 11  infrastructure_lat                 15718 non-null  flo

In [16]:
df_final['district'].unique()

array(['0', 'Адмиралтейский район', 'Василеостровский район',
       'Выборгский район', 'Калининский район', 'Кировский район', nan,
       'Красногвардейский район', 'Красносельский район',
       'Курортный район', 'Московский район', 'Невский район',
       'Петроградский район', 'Приморский район', 'Пушкинский район',
       'Фрунзенский район'], dtype=object)

In [17]:
df_final['region'].unique()

array(['Ленинградская область', 'Санкт-Петербург', nan], dtype=object)

In [18]:
df_final = df_final[df_final['region'] == 'Санкт-Петербург'].reset_index(drop=True)

In [19]:
df_final['region'].unique()

array(['Санкт-Петербург'], dtype=object)

In [20]:
df_final['district'].unique()

array(['0', 'Адмиралтейский район', 'Василеостровский район',
       'Выборгский район', 'Калининский район', 'Кировский район',
       'Красногвардейский район', 'Красносельский район',
       'Курортный район', 'Московский район', 'Невский район',
       'Петроградский район', 'Приморский район', 'Пушкинский район',
       'Фрунзенский район'], dtype=object)

In [21]:
df_final = df_final[df_final['district'] != '0'].reset_index(drop=True)

In [22]:
df_final['district'].unique()

array(['Адмиралтейский район', 'Василеостровский район',
       'Выборгский район', 'Калининский район', 'Кировский район',
       'Красногвардейский район', 'Красносельский район',
       'Курортный район', 'Московский район', 'Невский район',
       'Петроградский район', 'Приморский район', 'Пушкинский район',
       'Фрунзенский район'], dtype=object)

In [23]:
df_final = df_final.dropna(subset=['sport_object_name']).reset_index(drop=True)
df_final

sport_object_id       sport_object_name  \
0         5.348553e+15    Нептун, фитнес-центр   
1         5.348553e+15    Нептун, фитнес-центр   
2         5.348553e+15    Нептун, фитнес-центр   
3         5.348553e+15    Нептун, фитнес-центр   
4         5.348553e+15    Нептун, фитнес-центр   
...                ...                     ...   
15035     7.000000e+16  Padel star, падел клуб   
15036     7.000000e+16  Padel star, падел клуб   
15037     7.000000e+16  Padel star, падел клуб   
15038     7.000000e+16  Padel star, падел клуб   
15039     7.000000e+16  Padel star, падел клуб   

                   sport_object_address  sport_object_lat  sport_object_lon  \
0      набережная Обводного канала, 93а         59.916726          30.33898   
1      набережная Обводного канала, 93а         59.916726          30.33898   
2      набережная Обводного канала, 93а         59.916726          30.33898   
3      набережная Обводного канала, 93а         59.916726          30.33898   
4      набережная Обводного канала, 93а         59.916726          30.33898   
...                                 ...               ...               ...   
15035                   Фучика, 2 лит А         59.878109          30.35834   
15036                   Фучика, 2 лит А         59.878109          30.35834   
15037                   Фучика, 2 лит А         59.878109          30.35834   
15038                   Фучика, 2 лит А         59.878109          30.35834   
15039                   Фучика, 2 лит А         59.878109          30.35834   

      sport_object_type  sport_object_total_infrastructure  schedule  \
0                теннис                               93.0       0.0   
1                теннис                               93.0       0.0   
2                теннис                               93.0       0.0   
3                теннис                               93.0       0.0   
4                теннис                               93.0       0.0   
...                 ...                                ...       ...   
15035             падел                               82.0       0.0   
15036             падел                               82.0       0.0   
15037             падел                               82.0       0.0   
15038             падел                               82.0       0.0   
15039             падел                               82.0       0.0   

      infrastructure_type                                infrastructure_name  \
0                   метро  Петербургский метрополитен, служба материально...   
1                    кафе                                   Нептун, ресторан   
2                    кафе                                    Объект, кофейня   
3                    кафе                                     Аль-Шарк, кафе   
4                    кафе                                  Аригато, суши-бар   
...                   ...                                                ...   
15035                офис                            Континент, бизнес-центр   
15036                офис        Континент, торгово-развлекательный комплекс   
15037                офис                                           СберБанк   
15038                офис                                         Почта Банк   
15039                офис                      Бухарестская 24, бизнес-центр   

       ... Отклонение_плотности_%  Зарплата  Отклонение_зарплаты_%  Население  \
0      ...              77.841628  134590.0               1.376518   158253.0   
1      ...              77.841628  134590.0               1.376518   158253.0   
2      ...              77.841628  134590.0               1.376518   158253.0   
3      ...              77.841628  134590.0               1.376518   158253.0   
4      ...              77.841628  134590.0               1.376518   158253.0   
...    ...                    ...       ...                    ...        ...   
15035  ...              72.430749  119480.0             -10.0047

In [24]:
df_final.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 15040 entries, 0 to 15039
Data columns (total 34 columns):
 #   Column                             Non-Null Count  Dtype  
---  ------                             --------------  -----  
 0   sport_object_id                    15040 non-null  float64
 1   sport_object_name                  15040 non-null  object 
 2   sport_object_address               15040 non-null  object 
 3   sport_object_lat                   15040 non-null  float64
 4   sport_object_lon                   15040 non-null  float64
 5   sport_object_type                  15040 non-null  object 
 6   sport_object_total_infrastructure  15040 non-null  float64
 7   schedule                           15040 non-null  float64
 8   infrastructure_type                15040 non-null  object 
 9   infrastructure_name                15040 non-null  object 
 10  infrastructure_address             14193 non-null  object 
 11  infrastructure_lat                 15040 non-null  flo

In [25]:
df_final = df_final.drop(columns=['Тип_кластера_кластера','Порядковый_номер_в_кластере','Населенный_пункт','region'])
df_final

sport_object_id       sport_object_name  \
0         5.348553e+15    Нептун, фитнес-центр   
1         5.348553e+15    Нептун, фитнес-центр   
2         5.348553e+15    Нептун, фитнес-центр   
3         5.348553e+15    Нептун, фитнес-центр   
4         5.348553e+15    Нептун, фитнес-центр   
...                ...                     ...   
15035     7.000000e+16  Padel star, падел клуб   
15036     7.000000e+16  Padel star, падел клуб   
15037     7.000000e+16  Padel star, падел клуб   
15038     7.000000e+16  Padel star, падел клуб   
15039     7.000000e+16  Padel star, падел клуб   

                   sport_object_address  sport_object_lat  sport_object_lon  \
0      набережная Обводного канала, 93а         59.916726          30.33898   
1      набережная Обводного канала, 93а         59.916726          30.33898   
2      набережная Обводного канала, 93а         59.916726          30.33898   
3      набережная Обводного канала, 93а         59.916726          30.33898   
4      набережная Обводного канала, 93а         59.916726          30.33898   
...                                 ...               ...               ...   
15035                   Фучика, 2 лит А         59.878109          30.35834   
15036                   Фучика, 2 лит А         59.878109          30.35834   
15037                   Фучика, 2 лит А         59.878109          30.35834   
15038                   Фучика, 2 лит А         59.878109          30.35834   
15039                   Фучика, 2 лит А         59.878109          30.35834   

      sport_object_type  sport_object_total_infrastructure  schedule  \
0                теннис                               93.0       0.0   
1                теннис                               93.0       0.0   
2                теннис                               93.0       0.0   
3                теннис                               93.0       0.0   
4                теннис                               93.0       0.0   
...                 ...                                ...       ...   
15035             падел                               82.0       0.0   
15036             падел                               82.0       0.0   
15037             падел                               82.0       0.0   
15038             падел                               82.0       0.0   
15039             падел                               82.0       0.0   

      infrastructure_type                                infrastructure_name  \
0                   метро  Петербургский метрополитен, служба материально...   
1                    кафе                                   Нептун, ресторан   
2                    кафе                                    Объект, кофейня   
3                    кафе                                     Аль-Шарк, кафе   
4                    кафе                                  Аригато, суши-бар   
...                   ...                                                ...   
15035                офис                            Континент, бизнес-центр   
15036                офис        Континент, торгово-развлекательный комплекс   
15037                офис                                           СберБанк   
15038                офис                                         Почта Банк   
15039                офис                      Бухарестская 24, бизнес-центр   

       ... Плотность_населения  Отклонение_плотности_%  Зарплата  \
0      ...            11451.01               77.841628  134590.0   
1      ...            11451.01               77.841628  134590.0   
2      ...            11451.01               77.841628  134590.0   
3      ...            11451.01               77.841628  134590.0   
4      ...            11451.01               77.841628  134590.0   
...    ...                 ...                     ...       ...   
15035  ...            11102.61               72.430749  119480.0   
15036  ...            11102.61               72.430749  119480.0   
15037  ...         

In [26]:
# Сохраняем данные в CSV для использования в дашборд

df_final.to_csv('sport_objects_final_full_data.csv', index=False, encoding='utf-8-sig')

# Гипотезы:

### 1) Районы с более высокой плотностью населения имеют лучшее покрытие спортивными объектами.

### 2) Районы с более высокой средней зарплатой имеют лучшее покрытие спортивными объектами.

### 3) Районы с более высокой плотностью объектов инфраструктуры (кафе, ТЦ, супермаркеты, офисы, остановки и тд) имеют большее количество спортивных объектов.

### 4) Районы с высоким соотношением мужчин к женщинам (Соотношение_М_Ж) могут иметь больше количество спортивных объектов.



# Аналитика по гипотезам

In [27]:
# Группируем данные по районам для анализа

district_stats = df_final.groupby('district').agg({
    'sport_object_id': 'nunique',
    'sport_object_type': 'nunique',
    'Плотность_населения': 'first',
    'Зарплата': 'first',
    'Соотношение_М_Ж': 'first',
    'Кластер': 'first',
    'Класс_плотности': 'first',
    'Класс_зарплаты': 'first'
}).reset_index()

district_stats = district_stats.rename(columns={
    'sport_object_id': 'sport_objects_count',
    'sport_object_type': 'sport_types_count'
})
district_stats

district  sport_objects_count  sport_types_count  \
0      Адмиралтейский район                    3                  2   
1    Василеостровский район                    2                  2   
2          Выборгский район                    7                  2   
3         Калининский район                    2                  1   
4           Кировский район                    4                  3   
5   Красногвардейский район                    4                  2   
6      Красносельский район                    1                  1   
7           Курортный район                    2                  2   
8          Московский район                   10                  3   
9             Невский район                    5                  2   
10      Петроградский район                    7                  2   
11         Приморский район                    9                  3   
12         Пушкинский район                    2                  2   
13        Фрунзенский район                    3                  2   

    Плотность_населения  Зарплата  Соотношение_М_Ж  Кластер Класс_плотности  \
0              11451.01  134590.0            0.828      2.0   ОЧЕНЬ ВЫСОКАЯ   
1              12558.86  148492.0            0.816      2.0   ОЧЕНЬ ВЫСОКАЯ   
2               4620.91  131990.0            0.812      0.0         ВЫСОКАЯ   
3              13611.45  145695.0            0.813      2.0   ОЧЕНЬ ВЫСОКАЯ   
4               7162.79  129717.0            0.837      0.0         ВЫСОКАЯ   
5               6554.52  126337.0            0.795      0.0         ВЫСОКАЯ   
6               4711.81  108531.0            0.804      0.0         ВЫСОКАЯ   
7                308.41  118117.0            0.887      1.0          НИЗКАЯ   
8               4655.63  163332.0            0.776      3.0         СРЕДНЯЯ   
9               9008.82  120834.0            0.818      0.0         ВЫСОКАЯ   
10              5987.87  154086.0            0.946      4.0         СРЕДНЯЯ   
11              6303.48  166640.0            0.834      3.0         СРЕДНЯЯ   
12              1065.04  110107.0            0.878      1.0          НИЗКАЯ   
13             11102.61  119480.0            0.796      0.0         ВЫСОКАЯ   

   Класс_зарплаты  
0    ВЫШЕ СРЕДНЕЙ  
1    ВЫШЕ СРЕДНЕЙ  
2         СРЕДНЯЯ  
3    ВЫШЕ СРЕДНЕЙ  
4         СРЕДНЯЯ  
5         СРЕДНЯЯ  
6         СРЕДНЯЯ  
7         СРЕДНЯЯ  
8         ВЫСОКАЯ  
9         СРЕДНЯЯ  
10        ВЫСОКАЯ  
11        ВЫСОКАЯ  
12        СРЕДНЯЯ  
13        СРЕДНЯЯ

In [29]:
# Рассчитываем плотность инфраструктуры по районам

# Количество объектов инфраструктуры по районам
infra_by_district = df_final.groupby('district')['infrastructure_id'].nunique().reset_index()
infra_by_district = infra_by_district.rename(columns={'infrastructure_id': 'infrastructure_count'})

# Объединяем с основными данными
district_stats = pd.merge(district_stats, infra_by_district, on='district', how='left')

# Рассчитываем плотность (инфраструктура на 1 спортивный объект)
district_stats['infrastructure_per_sport_object'] = (
    district_stats['infrastructure_count'] / district_stats['sport_objects_count']
).round(2)  # округлим

In [32]:
# Рассчитываем показатели на душу населения
district_stats['sport_objects_per_100k'] = (
    district_stats['sport_objects_count'] / district_stats['Население'] * 100000
).round(2)

district_stats['infrastructure_per_100k'] = (
    district_stats['infrastructure_count'] / district_stats['Население'] * 100000
).round(2)

# Выводим статистику по подготовленным данным
print(f"Среднее количество спортивных объектов на район: {district_stats['sport_objects_count'].mean():.1f}")
print(f"Среднее количество видов спорта на район: {district_stats['sport_types_count'].mean():.1f}")


Подготовлено данных по районам: 14
Среднее количество спортивных объектов на район: 4.4
Среднее количество видов спорта на район: 2.1


# ГИПОТЕЗА 1

## Районы с более высокой плотностью населения имеют лучшее покрытие спортивными объектами

In [46]:
import plotly.graph_objects as go
import plotly.express as px

In [79]:
numeric_cols = ['Плотность_населения', 'Зарплата', 'Соотношение_М_Ж',
                'sport_objects_count', 'sport_objects_per_100k']

corr_matrix = district_stats[numeric_cols].corr()

fig = go.Figure(data=go.Heatmap(
    z=corr_matrix.values,
    x=corr_matrix.columns,
    y=corr_matrix.index,
    colorscale='RdYlBu',
    text=corr_matrix.round(2).values,
    texttemplate='%{text}'
))

fig.update_layout(height=800, width=800, title='Тепловая карта')
fig.show()

### Коэффициент корреляции: -0.18
### ОТРИЦАТЕЛЬНАЯ связь
### Гипотеза опровергнута

Детали:
1) Самый плотно населенный район: Калининский район

   Плотность: 13,611 чел/км²

   Спортивных объектов: 2


2) Самый малонаселенный район: Курортный район

   Плотность: 308 чел/км²

   Спортивных объектов: 2


3) Район с наибольшим количеством объектов: Московский район

   Спортивных объектов: 10

   Плотность населения: 4,656 чел/км²


4. В среднем на район:

   Плотность: 7,079 чел/км²

   Спортивных объектов: 4.4

# ГИПОТЕЗА 2

## Районы с более высокой средней зарплатой имеют лучшее покрытие спортивными объектами.

### Коэффициент корреляции: 0.71 -- смотреть хитмэп выше
### ОЧЕНЬ СИЛЬНАЯ положительная
### Гипотеза подтверждается

Детали:
1) Район с самой высокой зарплатой: Приморский район

   Средняя зарплата: 166,640 руб

   Спортивных объектов: 9

   Плотность населения: 6,303 чел/км


2) Район с самой низкой зарплатой: Красносельский район

   Средняя зарплата: 108,531 руб

   Спортивных объектов: 1

   Плотность населения: 4,712 чел/км


3) СРЕДНИЕ ПОКАЗАТЕЛИ:

   Средняя зарплата: 134,139 руб

   Среднее количество объектов: 4.4

# ГИПОТЕЗА 3

## Районы с более высокой плотностью объектов инфраструктуры (кафе, ТЦ, супермаркеты, офисы, остановки и тд) имеют большее количество спортивных объектов.

In [80]:
infra_cols = ['infrastructure_count', 'infrastructure_per_sport_object',
              'sport_objects_count', 'Плотность_населения', 'Зарплата',
              'Соотношение_М_Ж', 'sport_objects_per_100k']

corr_matrix = district_stats[infra_cols].corr()

fig = go.Figure(data=go.Heatmap(
    z=corr_matrix.values,
    x=corr_matrix.columns,
    y=corr_matrix.index,
    colorscale='RdYlBu',
    text=corr_matrix.round(2).values,
    texttemplate='%{text}'
))

fig.update_layout(height=800, width=800, title='Тепловая карта')
fig.show()

### Коэффициент корреляции: 0.93
### ОЧЕНЬ СИЛЬНАЯ положительная
### Гипотеза подтверждается

Детали:

Топ-5 районов по инфраструктуре на 1 спорт.объект:

1) Адмиралтейский район

2) Василеостровский район

3) Московский район

4) Фрунзенский район

5) Красносельский район


Топ-3 районов с наименьшей инфраструктурой вокруг спорт объектов:

1) Курортный район

2) Пушкинский район

3) Невский район

Самый сильный фактор: Инфраструктура (0.93)

# ГИПОТЕЗА 4

## Районы с высоким соотношением мужчин к женщинам (Соотношение_М_Ж) могут иметь больше количество спортивных объектов.

### Коэффициент корреляции: -0.04 -- смотреть хитмэп выше
### ОЧЕНЬ СЛАБАЯ или отсутствует связь
### Гипотеза опровергнута

Детали:

Корреляция практически равна нулю.
Это означает, что соотношение мужчин и женщин в районе
НЕ ВЛИЯЕТ на количество спортивных объектов.